lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -839,6 +851,54 @@ class DecomposeAtenMeanOp : public OpRewritePattern<AtenMeanOp> {
 };
 } // namespace
 
+// productDimSize = product(size(dim) for dim in dims)
+// aten.mean(x, dims) = aten.sum(x, dims) / productDimSize.
+namespace {
+class DecomposeAtenMeanDimOp : public OpRewritePattern<AtenMeanDimOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenMeanDimOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    Value input = op.self();
+    Value dimList = op.dim();
+    Value keepDim = op.keepdim();
+    Value dtype = op.dtype();
+    Type outputType = op.getType();
+    MLIRContext *context = op.getContext();
+
+    BaseTensorType inputType = input.getType().cast<BaseTensorType>();
+    if (!inputType.hasDtype() || !inputType.getDtype().isa<mlir::FloatType>() ||
+        !isNoneOrFloatDtype(context, dtype)) {
+      return rewriter.notifyMatchFailure(
+          op, "only floating-point type is supported");
```

Comment:
Just out of curiosity, what's our rationale behind choosing between `notifyMatchFailure()` versus `emitError()`?  Should we use `emitError()` here, so that the end user does not need to re-run the compilation with the `--debug` flag?